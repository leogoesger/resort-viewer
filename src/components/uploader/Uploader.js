import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import {setPropTypes} from 'recompose';

const Uploader = ({onUpload}) => {
  return (
    <Button variant="raised" color="primary" component="label">
      {'Upload'}
      <input
        onChange={e => onUpload(e.target.files)}
        style={{display: 'none'}}
        type="file"
      />
    </Button>
  );
};

export default setPropTypes({onUpload: PropTypes.func.isRequired})(Uploader);
