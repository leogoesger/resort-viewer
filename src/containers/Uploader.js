import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import csv from 'csvtojson';
import {compose, setPropTypes, lifecycle} from 'recompose';

import UploaderHOC from '../hocs/UploaderHOC';
import {uploadResorts} from '../actions/resort';

const stringProcessor = (csvStr, cb) => {
  csv({})
    .fromString(csvStr)
    .on('err', err => cb(err, null))
    .then(csvRows => cb(null, csvRows));
};

const Uploader = UploaderHOC(stringProcessor);

const mapDispatchToProps = dispatch => {
  return {uploadResorts: d => dispatch(uploadResorts(d))};
};

const composer = mapDispatchToProps => {
  return compose(
    connect(
      null,
      mapDispatchToProps
    ),
    setPropTypes({
      uploadResorts: PropTypes.func.isRequired,
    }),
    lifecycle({
      componentDidMount() {
        document.title = 'UpMetric | Upload';
      },
    })
  );
};

export default composer(mapDispatchToProps)(Uploader);
