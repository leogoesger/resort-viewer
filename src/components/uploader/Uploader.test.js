import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Uploader from './Uploader';

describe('<Uploader />', () => {
  it('Render Uploader', () => {
    const wrapper = shallow(<Uploader onUpload={jest.fn()} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
