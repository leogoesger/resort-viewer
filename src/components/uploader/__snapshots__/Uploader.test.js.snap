// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Uploader /> Render Uploader 1`] = `
<WithStyles(Button)
  color="primary"
  component="label"
  variant="raised"
>
  Upload
  <input
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</WithStyles(Button)>
`;
